### ISSUE TITLE
False flag of `reportUnnecessaryComparison` on match case with `int(<float>)`

### ISSUE THREAD

**Issue by q-wertz** [DetachHead/basedpyright#1565]
### Description

In the following code
```python
sr = int(2.0e6)

match sr:
    case int(2.0e6):
        a = "asdf"
    case int(2.4e6):
        a = "cvbn"
    case _:
        raise ValueError

print(a)
```
a `reportUnnecessaryComparison` error is shown ([Playground](https://basedpyright.com/?typeCheckingMode=all&code=M4JwBAvGCWB2AuAKATAOgAwFMBsBKAUPgLYCG8AxgBZigBc%2BYjY5JwmMCKGOu9T-YEpDAAiVgBMAZiIZMWbDkjQAWHnwGMhUEeQBuAI1gz%2B89gH11GkCWgKAaiQA2AV0wBRECAD2IQgAcQOCQSXCA)).

In pyright no error is raised ([Playground with similar settings to basedpyright](https://pyright-play.net/?code=M4JwBAvGCWB2AuAKATAOgAwFMBsBKAUPgLYCG8AxgBZigBc%2BYjY5JwmMCKGOu9T-YEpDAAiVgBMAZiIZMWbDkjQAWHnwGMhUEeQBuAI1gz%2B89gH11GkCWgKAaiQA2AV0wBRECAD2IQgAcQOCQSXCA)).

As soon as it is replaced by integers directly the error in the correct case branch vanishes.
```python
sr = 2000000

match sr:
    case 2000000:
        a = "asdf"
    case 2400000:
        a = "cvbn"
    case _:
        raise ValueError

print(a)
```

Not sure if some float/int thing I don't get right but I think it also should not fail in the first example for the first branch.

**Comment by DetachHead** [comment]
[this does happen in pyright as well](https://pyright-play.net/?reportUnnecessaryComparison=true&code=M4JwBAvGCWB2AuAKATAOgAwFMBsBKAUPgLYCG8AxgBZigBc%2BYjY5JwmMCKGOu9T-YEpDAAiVgBMAZiIZMWbDkjQAWHnwGMhUEeQBuAI1gz%2B89gH11GkCWgKAaiQA2AV0wBRECAD2IQgAcQOCQSXCA), that playground link just didn't have the rule enabled

**Comment by q-wertz** [comment]
Ah sorry. I just used the link from the issue description and didn't think more about it 🙈 Maybe I also accidentally switched something off 😅 

Should I then report it upstream?

**Comment by DetachHead** [comment]
> Ah sorry. I just used the link from the issue description and didn't think more about it 🙈 Maybe I also accidentally switched something off 😅

hmm, there seems to be a bug in the upstream version of the playground that causes it to delete the settings from the url...

> Should I then report it upstream?

up to you. might get fixed quicker if you do

**Comment by q-wertz** [comment]
Reported [upstream](https://github.com/microsoft/pyright/issues) in issue 11026
Will be fixed in next release

**Comment by q-wertz** [comment]
Thanks a lot 😊
